### ISSUE TITLE
Update logic for retrieving page data from subfolders

### ISSUE THREAD

**Issue by alexanderleegs** [isomerpages/isomercms-frontend#361]
Currently, due to the navigation file revamp, our existing method of retrieving file information does not work due to the file path containing slashes. To fix this, we should modify the endpoints to make use of encode/decodeURIComponent, to allow path names to be handled properly.

**Comment by kwajiehao** [comment]
Adding the following for more clarity:

### Problem

Currently, we can't access subfolder pages through the `EditPage` layout. This is because of the changes we've made to our Isomer repos when we migrated to Jekyll v4 - third-nav sections/subfolders are now stored in actual sub-directories, whereas before all pages in a collection/folder were flat (no hierarchy). An example of a collection directory before (where `1a-page2.md` and `1b-page3.md` are in the same third-nav section, as seen by the shared number prefix in `1a` and `1b`):

```
Collection A
- 0-page1.md
- 1a-page2.md
- 1b-page3.md
```

However, with the new repo structure, this collection would no longer be flat:

```
Collection A
- page1.md
- subfolder 1
    - page2.md
    - page3.md
```

Therefore, when we access these folders, the URL ends with `<collection name>/<subfolder name>/<page name>`, vs before, when it was simply `<collection name>/<page name>`. To solve this issue, we need to URI encode `<subfolder name>/<page name>` to fit our current frontend routing.

**Comment by kwajiehao** [comment]
Fixed with PR #379
